### DISCUSSION TITLE
Software used for designing diagrams in this tutorial ?

### DISCUSSION THREAD

**Discussion by 3sigma** [alexzhornyak/SCXML-tutorial#8] [Q&A] [answered]
Hello,

It is not an issue, I just would like to know which software you used to design the diagrams shown in this SCXML tutorial.

Thanks in advance,

Nicolas

**Comment by alexzhornyak** [comment, marked as answer]
It was made using [**Scxml Editor**](https://github.com/alexzhornyak/ScxmlEditor-Tutorial).

The example of how this program looks like you can find **[here](https://youtu.be/tL-NICRwggY)**

**Comment by 3sigma** [comment]
Thank you. But do you have more details about this software (who made it, where can it be downloaded ?) because I didn't not find anything ressembling to it during my searches on the Internet.

Thanks in advance,

Nicolas

**Comment by alexzhornyak** [comment]
You can find portable version **[here](https://www.dropbox.com/sh/fjzm9ejdrtra1c0/AAB_ASgIPRFLX57x7rWPEv3Ta?dl=0)**

System requirements: Win 7,8,10

Help will be provided later

**Comment by alexzhornyak** [comment]
Small help can be found [here](https://github.com/alexzhornyak/ScxmlEditor-Tutorial)

**Comment by stil** [comment]
Is the software open source? I'm not comfortable using binaries downloaded from Dropbox.

**Comment by alexzhornyak** [comment]
> Is the software open source? I'm not comfortable using binaries downloaded from Dropbox.

I completely understand you in the questions of security.
Unfortunately [ScxmlEditor](https://github.com/alexzhornyak/ScxmlEditor-Tutorial) is not an open source now because it uses non-open source solutions. ScxmlEditor team is too small to migrate source code to some opensource framework.

I wish I could make it opensource and cross-platform but I do not have an opportunity to complete this goal **without sponsors and a team** because a lot of code should be overwritten and replaced with opensource libs.

**P.S. Concerning the questions of security:** ScxmlEditor doesn't collect or send any data. All activity is inside its app folder. It doesn't use any connections until you start a debugging. For debugging it opens 1 UDP port for sending messages to testing application and 1 UDP port for receiving. If you are debugging with [BasicHttp enabled](https://alexzhornyak.github.io/SCXML-tutorial/Doc/BasicHTTPEventIO.html) it uses 1 Http listen port and 1 Http web socket port.

![image](https://user-images.githubusercontent.com/18611095/105642335-d7e04480-5e91-11eb-8fe8-548d7bc1570f.png)

**Comment by alexzhornyak** [comment]
[Source code of ScxmlEditor](https://github.com/alexzhornyak/ScxmlEditor-Tutorial/tree/master/Src) is available now
